aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/Translator.java

```diff
@@ -111,35 +113,22 @@ static UntagLogGroupRequest translateToUntagLogGroupRequest(final String logGrou
                 .build();
     }
 
-    static ResourceModel translateForRead(final DescribeLogGroupsResponse response, final ListTagsLogGroupResponse tagsResponse) {
-        final String logGroupName = streamOfOrEmpty(response.logGroups())
-                .map(software.amazon.awssdk.services.cloudwatchlogs.model.LogGroup::logGroupName)
-                .filter(Objects::nonNull)
-                .findAny()
-                .orElse(null);
-        final String logGroupArn = streamOfOrEmpty(response.logGroups())
-                .map(software.amazon.awssdk.services.cloudwatchlogs.model.LogGroup::arn)
-                .filter(Objects::nonNull)
-                .findAny()
-                .orElse(null);
-        final Integer retentionInDays = streamOfOrEmpty(response.logGroups())
-                .map(software.amazon.awssdk.services.cloudwatchlogs.model.LogGroup::retentionInDays)
-                .filter(Objects::nonNull)
-                .findAny()
-                .orElse(null);
-        final String kmsKeyId = streamOfOrEmpty(response.logGroups())
-                .map(software.amazon.awssdk.services.cloudwatchlogs.model.LogGroup::kmsKeyId)
-                .filter(Objects::nonNull)
+    static ResourceModel translateForRead(final DescribeLogGroupsResponse response, final ListTagsLogGroupResponse tagsResponse, final String requestLogGroupName) {
+        LogGroup matchedLogGroup = streamOfOrEmpty(response.logGroups())
+                .filter(logGroup -> logGroup.logGroupName() != null && logGroup.logGroupName().equals(requestLogGroupName))
                 .findAny()
                 .orElse(null);
         final Set<Tag> tags = translateSdkToTags(Optional.ofNullable(tagsResponse)
                 .map(ListTagsLogGroupResponse::tags)
                 .orElse(null));
+        if (matchedLogGroup == null) return ResourceModel.builder()
+                .tags(tags)
+                .build();
         return ResourceModel.builder()
-                .arn(logGroupArn)
-                .logGroupName(logGroupName)
-                .retentionInDays(retentionInDays)
-                .kmsKeyId(kmsKeyId)
+                .arn(matchedLogGroup.arn())
```

Comment:
possible NPE for all of these. I'd split up so that 

if (null), return a null model
else, model.arn(lg.arn).etc

Reply:
On line 124 I checked that `matchedLogGroup` is not null, wont the getter functions just return null if it is in fact null?

Reply:
oh didn't see the return, small style nit is not to have inline returns.